helix-tui/src/buffer.rs

```diff
@@ -278,6 +290,11 @@ impl Buffer {
     where
         S: AsRef<str>,
     {
+        // prevent panic if out of range
+        if self.index_of_opt(x, y).is_none() || width == 0 {
+            return (x, y);
+        }
```

Comment:
This is now doing the math twice, once in `index_of_opt` -> `index_of`, then once on L298.

Let's extract an `in_bounds(x, y) -> bool` method and use that here.